collector/index.md

```diff
@@ -0,0 +1,425 @@
+    </tr>
+    <tr>
+      <td>aws_db_instance_id (AWS_INSTANCE_ID)</td>
+      <td>n/a, required if aws</td>
+      <td>the instance id of your AWS server</td>
+    </tr>
+    <tr>
+      <td>aws_access_key_id (AWS_ACCESS_KEY_ID)</td>
+      <td>n/a, required if aws and using logs?</td>
+      <td>access key id of IAM role with permissions to?</td>
+    </tr>
+    <tr>
+      <td>aws_secret_access_key (AWS_SECRET_ACCESS_KEY)</td>
+      <td>n/a, required if aws and using logs?</td>
+      <td>secret access key of IAM role with permissions to?</td>
+    </tr>
+    <tr>
+      <td>aws_endpoint_signing_region (AWS_ENDPOINT_SIGNING_REGION)</td>
+      <td>[none]</td>
+      <td>?</td>
+    </tr>
+    <tr>
+      <td>aws_endpoint_rds_url (AWS_ENDPOINT_RDS_URL)</td>
+      <td>[none]</td>
+      <td>?</td>
+    </tr>
+    <tr>
+      <td>aws_endpoint_ec2_url (AWS_ENDPOINT_EC2_URL)</td>
+      <td>[none]</td>
+      <td>?</td>
+    </tr>
+    <tr>
+      <td>aws_endpoint_cloudwatch_url (AWS_ENDPOINT_CLOUDWATCH_URL)</td>
+      <td>[none]</td>
+      <td>?</td>
+    </tr>
+    <tr>
+      <td>aws_endpoint_cloudwatch_logs_url (AWS_ENDPOINT_CLOUDWATCH_LOGS_URL)</td>
+      <td>[none]</td>
+      <td>?</td>
+    </tr>
+  </tbody>
+</table>
+
+## Azure settings
+
+Only relevant if you are running your database in Azure.
+
+<table>
+  <thead>
+    <tr>
+      <th>setting</th>
+      <th>default</th>
+      <th>description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>azure_db_server_name (AZURE_DB_SERVER_NAME)</td>
+      <td>n/a, required if azure?</td>
+      <td>?</td>
+    </tr>
+    <tr>
+      <td>azure_eventhub_namespace (AZURE_EVENTHUB_NAMESPACE)</td>
+      <td>n/a, required if azure?</td>
+      <td>?</td>
+    </tr>
+    <tr>
+      <td>azure_eventhub_name (AZURE_EVENTHUB_NAME)</td>
+      <td>n/a, required if azure?</td>
+      <td>?</td>
+    </tr>
+    <tr>
+      <td>azure_ad_tenant_id (AZURE_AD_TENANT_ID)</td>
+      <td>n/a, required if azure?</td>
+      <td>?</td>
+    </tr>
+    <tr>
+      <td>azure_ad_client_id (AZURE_AD_CLIENT_ID)</td>
+      <td>n/a, required if azure?</td>
+      <td></td>
+    </tr>
+    <tr>
+      <td>azure_ad_client_secret (AZURE_AD_CLIENT_SECRET)</td>
+      <td>n/a, required if azure?</td>
+      <td></td>
+    </tr>
+    <tr>
+      <td>azure_ad_certificate_path (AZURE_AD_CERTIFICATE_PATH)</td>
+      <td>n/a, required if azure?</td>
+      <td></td>
+    </tr>
+    <tr>
+      <td>azure_ad_certificate_password (AZURE_AD_CERTIFICATE_PASSWORD)</td>
+      <td>n/a, required if azure?</td>
+      <td></td>
+    </tr>
+  </tbody>
+</table>
+
+## Google Cloud Platform
+
+Only relevant if you are running your database in GCP.
+
+<table>
+  <thead>
+    <tr>
+      <th>setting</th>
+      <th>default</th>
+      <th>description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>gcp_cloudsql_instance_id (GCP_CLOUDSQL_INSTANCE_ID)</td>
+      <td>n/a, required if gcp?</td>
```

Comment:
This may be auto-detected from hostname as well (not sure), double check the code.

Reply:
Does not look like it, but then I don't see where this happens for AWS either. I searched for [AwsDbInstanceID](https://github.com/pganalyze/collector/search?q=AwsDbInstanceID&unscoped_q=AwsDbInstanceID) and none of those look relevant. :confused:

Reply:
AWS and Azure happen here: 

https://github.com/pganalyze/collector/blob/8bd5b1b56df9423b10bdf284d636fe6ac5e153e7/config/read.go#L166

GCP uses IP addresses instead of hostnames, so the logic doesn't apply there - thus this is required when using Log Insights (and in general recommended, since uniquely identifies the server)